File: aplicação/backend/src/main/java/eaproject/output/GetResultByIdOutput.java
package eaproject.output;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.io.Serializable;
import java.sql.Timestamp;

public class GetResultByIdOutput extends BaseOutput implements Serializable {

    @JsonProperty
    private ResultProperties result;

    public ResultProperties getResult() {
        return result;
    }

    public void setResult(ResultProperties result) {
        this.result = result;
    }

    public static class ResultProperties implements Serializable {
        private int id;
        private int position;
        private Timestamp time;
        private String observations;
        private Timestamp penaltyTime;
        private Timestamp creationDate;

        // Getters and setters
        public int getId() {
            return id;
        }

        public void setId(int id) {
            this.id = id;
        }

        public int getPosition() {
            return position;
        }

        public void setPosition(int position) {
            this.position = position;
        }

        public Timestamp getTime() {
            return time;
        }

        public void setTime(Timestamp time) {
            this.time = time;
        }

        public String getObservations() {
            return observations;
        }

        public void setObservations(String observations) {
            this.observations = observations;
        }

        public Timestamp getPenaltyTime() {
            return penaltyTime;
        }

        public void setPenaltyTime(Timestamp penaltyTime) {
            this.penaltyTime = penaltyTime;
        }

        public Timestamp getCreationDate() {
            return creationDate;
        }

        public void setCreationDate(Timestamp creationDate) {
            this.creationDate = creationDate;
        }
    }
}

File: aplicação/frontend/ea-project/src/models/output/GetResultByIdOutput.js
import BaseOutput from "./BaseOutput";

export default class GetResultByIdOutput extends BaseOutput {
  constructor(
    creationDate = new Date(),
    penaltyTime = new Date(),
    observations = "",
    time = new Date(),
    position = 0,
    id = 0,
    feedbackMessages = []
  ) {
    super(feedbackMessages);
    this.id = id;
    this.position = position;
    this.time = time;
    this.observations = observations;
    this.penaltyTime = penaltyTime;
    this.creationDate = creationDate;
  }
}
